src/mds/MDLog.cc

```diff
@@ -253,37 +253,22 @@ void MDLog::append()
 
 // -------------------------------------------------
 
-void MDLog::_start_entry(LogEvent *e)
+void MDLog::_submit_entry(LogEvent *le, MDSLogContextBase* c)
```

Comment:
So, with change now more than one event can be submitted followed by calling `_segment_upkeep()` whereas earlier a `_submit_entry()` _always_ followed a corresponding `_start_entry()`? IOW, this batches up journal events.

Reply:
Not sure I understand. Before you had to:

```
start_entry(e)
submit_entry(e)
```

now it's just:

```
submit_entry(e)
```

There's no batching at all.

Reply:
Right, we're indirectly relying on the mds_lock to serialize the creation/submission of events.  (We've always done so; this old API is technically racy and doesn't help with that.)